src/emulator/auth/operations.ts

```diff
@@ -1739,31 +1740,25 @@ function getEmulatorProjectConfig(state: ProjectState): Schemas["EmulatorV1Proje
 
 function updateEmulatorProjectConfig(
   state: ProjectState,
-  reqBody: Schemas["EmulatorV1ProjectsConfig"]
+  reqBody: Schemas["EmulatorV1ProjectsConfig"],
+  ctx: ExegesisContext
 ): Schemas["EmulatorV1ProjectsConfig"] {
-  const allowDuplicateEmails = reqBody.signIn?.allowDuplicateEmails;
-  if (allowDuplicateEmails != null) {
-    assert(
-      state instanceof AgentProjectState,
-      "((Only top level projects can set oneAccountPerEmail.))"
-    );
-    state.oneAccountPerEmail = !allowDuplicateEmails;
-  }
-  const usageMode = reqBody.usageMode;
-  if (usageMode != null) {
-    assert(state instanceof AgentProjectState, "((Only top level projects can set usageMode.))");
-    switch (usageMode) {
-      case "PASSTHROUGH":
-        assert(state.getUserCount() === 0, "Users are present, unable to set passthrough mode");
-        state.usageMode = UsageMode.PASSTHROUGH;
-        break;
-      case "DEFAULT":
-        state.usageMode = UsageMode.DEFAULT;
-        break;
-      default:
-        throw new BadRequestError("Invalid usage mode provided");
+  // New developers should not use updateEmulatorProjectConfig to update the
+  // allowDuplicateEmails and usageMode settings and should instead use
+  // updateConfig to do so.
+  let updateMask = "";
+  if (reqBody.signIn?.allowDuplicateEmails) {
+    updateMask += "signIn.allowDupliateEmails";
+  }
+  if (reqBody.usageMode) {
+    assert(reqBody.usageMode !== "USAGE_MODE_UNSPECIFIED", "Invalid usage mode provided");
+    if (reqBody.usageMode === "PASSTHROUGH") {
+      assert(state.getUserCount() === 0, "Users are present, unable to set passthrough mode");
     }
+    updateMask += ",usageMode";
```

Comment:
This probably won't work if `allowDuplicateEmails` is not set while `usageMode` is set. Please use an Array and join with `,` before calling `updateConfig`.